File: 프로그래머스/2/12913. 땅따먹기/땅따먹기.java
class Solution {
    int solution(int[][] land) {
        for (int j = land.length - 2; j > -1; j--) {
            for (int i = 0; i < land[j].length; i++) {
                int max = 0;
                for (int k = 0; k < i; k++) {
                    max = (int) Math.max(max, land[j+1][k]);
                }
                
                for (int k = i + 1; k < land[j].length; k++) {
                    max = (int) Math.max(max, land[j+1][k]);
                }
                land[j][i] += max;
            }
        }
        
        int answer = 0;
        for (int l : land[0]) {
            if (answer < l) {
                answer = l;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/2/12913. 땅따먹기/땅따먹기.py
def solution(land):
    for j in range(len(land) - 2, -1 , -1):
        for i in range(len(land[j])):
            land[j][i] += max(land[j+1][:i] + land[j+1][i+1:])
    
    
    return max(land[0])
            
            
            
